CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatWindow.lua

```diff
@@ -0,0 +1,305 @@
+local source = [[
+local module = {}
+--////////////////////////////// Include
+--//////////////////////////////////////
+local modulesFolder = script.Parent
+local moduleChatChannel = require(modulesFolder:WaitForChild("ChatChannel"))
+local moduleTransparencyTweener = require(modulesFolder:WaitForChild("TransparencyTweener"))
+
+--////////////////////////////// Details
+--//////////////////////////////////////
+local metatable = {}
+metatable.__ClassName = "ChatWindow"
+
+metatable.__tostring = function(tbl)
+	return tbl.__ClassName .. ": " .. tbl.MemoryLocation
+end
+
+metatable.__metatable = "The metatable is locked"
+metatable.__index = function(tbl, index, value)
+	if rawget(tbl, index) then return rawget(tbl, index) end
+	if rawget(metatable, index) then return rawget(metatable, index) end
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+metatable.__newindex = function(tbl, index, value)
+	error(index .. " is not a valid member of " .. tbl.__ClassName)
+end
+
+
+--////////////////////////////// Methods
+--//////////////////////////////////////
+local function CreateGuiObject()
+	local BaseFrame = Instance.new("Frame")
+	BaseFrame.BackgroundTransparency = 1
+	BaseFrame.Size = UDim2.new(0.35, 0, 0.35, 0)
+
+	--// This spawns a new thread that waits approximately long enough 
+	--// until the BaseFrame has been parented to PlayerGui so it can 
+	--// actually check the size of it.
+	spawn(function()
+		wait()
+		if (BaseFrame.AbsoluteSize.X < 1600/3.75) then
+			BaseFrame.Size = UDim2.new(0, 1600/3.75, 0, 900/3.75)
+		end
+	end)
+
+	--// Chat Size18, 8 pixels on each end for white box in center, 
+	--// 8 pixels on each end for actual chatbot object
+	local chatBarYSize = 18 + 16 + 16
```

Comment:
Instead of hard-coding 18 here we should extract the height from the font size we are using (in-case we ever change the size or allow the user to customize their font size)